test/sys/test_ptrace.rs

```diff
@@ -55,6 +158,8 @@ fn test_ptrace_cont() {
     use nix::unistd::fork;
     use nix::unistd::ForkResult::*;
 
+    let _m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
```

Comment:
This is unrelated to all the other changes in this commit. Can you break this out into a separate commit?

Reply:
I simply removed that for now. This can be added once the changes are squashed.